v4/feature/receipt/src/main/java/exchange/dydx/trading/feature/receipt/components/isolatedmargin/DydxReceiptIsolatedPositionMarginUsageViewModel.kt

```diff
@@ -21,25 +28,83 @@ class DydxReceiptIsolatedPositionMarginUsageViewModel @Inject constructor(
+        pendingPosition: SubaccountPendingPosition?,
     ): DydxReceiptIsolatedPositionMarginUsageView.ViewState {
-        val marginUsage: TradeStatesWithDoubleValues? =
-            null
-        /*
-        TODO: After abacus exposes marginUsage, changes to next line
-        if (marketId != null) abacusStateManager.state.selectedSubaccountPositionOfMarket(marketId).value?.leverage else null
-         */
         return DydxReceiptIsolatedPositionMarginUsageView.ViewState(
             localizer = localizer,
             formatter = formatter,
-            before = marginUsage?.current,
-            after = marginUsage?.postOrder,
+            before = positionMarginCurrent(
+                marginMode = position?.marginMode,
+                equity = position?.equity,
+                notionalTotal = position?.notionalTotal,
+                maintenanceMarginFraction = position?.adjustedMmf?.current,
+            ) ?: positionMarginCurrent(
+                marginMode = MarginMode.Isolated,
+                equity = pendingPosition?.equity,
+                notionalTotal = null,
+                maintenanceMarginFraction = null,
+            ),
+            after = positionMarginPostOrder(
+                marginMode = position?.marginMode,
+                equity = position?.equity,
+                notionalTotal = position?.notionalTotal,
+                maintenanceMarginFraction = position?.adjustedMmf?.postOrder,
+            ) ?: positionMarginPostOrder(
+                marginMode = MarginMode.Isolated,
+                equity = pendingPosition?.equity,
+                notionalTotal = null,
+                maintenanceMarginFraction = null,
+            ),
         )
     }
+
+    private fun positionMarginCurrent(
+        marginMode: MarginMode?,
+        equity: TradeStatesWithDoubleValues?,
+        notionalTotal: TradeStatesWithDoubleValues?,
+        maintenanceMarginFraction: Double?
+    ): Double? {
+        return when (marginMode) {
+            MarginMode.Cross -> return notionalTotal?.current?.let { notionalTotal ->
+                maintenanceMarginFraction?.let { maintenanceMarginFraction ->
+                    maintenanceMarginFraction * notionalTotal
+                }
+            }
+            MarginMode.Isolated -> equity?.current
+            else -> null
+        }
+    }
+
+    private fun positionMarginPostOrder(
+        marginMode: MarginMode?,
+        equity: TradeStatesWithDoubleValues?,
+        notionalTotal: TradeStatesWithDoubleValues?,
+        maintenanceMarginFraction: Double?
+    ): Double? {
+        return when (marginMode) {
+            MarginMode.Cross -> return notionalTotal?.postOrder?.let { notionalTotal ->
+                maintenanceMarginFraction?.let { maintenanceMarginFraction ->
+                    maintenanceMarginFraction * notionalTotal
+                }
+            }
+            MarginMode.Isolated -> equity?.postOrder
+            else -> null
+        }
+    }
```

Comment:
these types of helper methods feel like they should be in abacus. (easy to make a mistake and needed on both platforms)

Reply:
Yes, already planned to move this to Abacus